content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,330 @@
+The encrypted secret can be safely stored in a public or private Git repository.
+
+The Kustomize Controller will pull the Kubernetes Secret manifest with encrypted values from the Source, (the metadata
+is stored in plain-text), then Kustomize Controller decrypts its encrypted values using the supplied key.
+
+Note, it's a good idea to also back up your secret values in a password manager or other form of secure external storage.
+
+The decrypted manifests are kept in memory and passed on to the next stage.
+
+### 9. Kustomize Controller (Build)
+
+Before it applies YAML or JSON resurce declarations to the Kubernetes API for its cluster, the Kustomize Controller
+reads the artifact files from its source path and builds them using the Kustomize Go library's `build` call. This call
+returns any custom resource definitions (CRDs), namespaces, or other cluster-wide resources it renders before
+subordinate custom resources or namespace-scoped resources so that they will be available in the API for the resources
+that refer to or use them.
+
+* FAQ: [How does Flux run `kustomize build` internally?][Kustomize Build Flags] (Replicate Kustomize behavior locally with the Kustomize CLI)
+
+### 10. Kustomize Controller (Server Side Apply)
+
+**Needs a link to FluxCD documentation for Server-Side Apply process!** _The only current on-point reference is the [Server-side reconciliation is coming](https://fluxcd.io/blog/2021/09/server-side-reconciliation-is-coming/) blog post.  This would likely be considered a gap in FluxCD project docs._
+
+The Kustomize Controller communicates directly with the Kubernetes API using
+[server-side apply and update][] API operations
+instead of running the `kubectl apply` command as a separate forked process and passing it manifest data through
+a system pipe. Applying resource manifests directly
+to the Kubernetes API is both more efficient and provides more control over the process, enabling the Kustomize
+Controller to give real-time feedback on validation errors, garbage-collection and resource health assessment.
+It also allows the Kubernetes API to track
+[field management][],
+so different management tools or controllers can set field values within the same resource without interfering
+with each other.
+
+The server-side apply operation is synchronous rather than asynchronous. If any resources fail to become ready before a
+specified timeout, the controller can abort the entire transaction.
+
+The Kustomize Controller applies resource manifests to match the order in which they were rendered by the kustomize `build`
+call.  It therefore applies any custom resource definition (CRD), namespace, or cluster-scoped resources before their subordinate
+custom resource or namespace-scoped resources to that they will be available in the API for the resources that refer to or use them.
+
+<insert sequence diagram>?
```

Comment:
(this is TODO 4/5)

Reply:
The diagrams are in here now, there are formatting issues but we can resolve them subsequently.

For some reason everything is `height="1024"` in the Chrome debugger/html source.

I'm not sure if that looks better on larger screens, but on my 13" MBP it's quite cramped and unnecessary whitespace on either side of the document. I imagine it does look better though, at 1080p or 4k resolutions and above. (So maybe better to leave it this way?)